### ISSUE TITLE
CodeGenerator.MSBuild v3.4.3 does not work with records

### ISSUE THREAD

**Issue by wdspider** [dotnet/orleans#7161]
Using this basic grain:
```csharp
    public sealed record LocationInfo
    {
        public static LocationInfo Default { get; } = new LocationInfo { Id = -1, Name = "Unknown Location" };

        public long Id { get; init; }
        public string Name { get; init; } = "";
    }

    public interface ILocationGrain : IGrainWithIntegerKey
    {
        public Task<LocationInfo> GetLocationInfoAsync();
        public Task SetInfoAsync(LocationInfo locationInfo);
    }

    public sealed class LocationGrain : Grain, ILocationGrain
    {
        #region Internal State
        private LocationInfo locationInfo = LocationInfo.Default;
        #endregion

        #region ILocationGrain Implementation
        Task<LocationInfo> ILocationGrain.GetLocationInfoAsync() => Task.FromResult(locationInfo);

        Task ILocationGrain.SetInfoAsync(LocationInfo info)
        {
            locationInfo = info;
            return Task.CompletedTask;
        }
        #endregion
    }
```

The code gen fails with these errors:
```
1>Orleans.CodeGenerator - command-line = SourceToSource C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.args.txt
1>EXEC : warning : attempted to get TypeSyntax for unknown (error) type, "LocationInfo". Possible reason: None. Possible candidates:
1>EXEC : warning : attempted to get TypeSyntax for unknown (error) type, "LocationInfo". Possible reason: None. Possible candidates:
1>EXEC : warning : attempted to get TypeSyntax for unknown (error) type, "LocationInfo". Possible reason: None. Possible candidates:
1>EXEC : warning : attempted to get TypeSyntax for unknown (error) type, "LocationInfo". Possible reason: None. Possible candidates:
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(102,45,102,57): error CS0246: The type or namespace name 'LocationInfo' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(102,118,102,138): error CS0539: 'OrleansCodeGenLocationGrainReference.GetLocationInfoAsync()' in explicit interface declaration is not found among members of the interface that can be implemented
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(107,117,107,129): error CS0246: The type or namespace name 'LocationInfo' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(107,104,107,116): error CS0539: 'OrleansCodeGenLocationGrainReference.SetInfoAsync(LocationInfo)' in explicit interface declaration is not found among members of the interface that can be implemented
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(57,99,57,157): error CS0535: 'OrleansCodeGenLocationGrainReference' does not implement interface member 'ILocationGrain.GetLocationInfoAsync()'
1>C:\Repos\src\server\Abstractions\obj\Debug\net6.0\Abstractions.orleans.g.cs(57,99,57,157): error CS0535: 'OrleansCodeGenLocationGrainReference' does not implement interface member 'ILocationGrain.SetInfoAsync(LocationInfo)'
1>Done building project "Abstractions.csproj" -- FAILED.
```

The above errors go away if I change `LocationInfo` to a class rather than a record.  In new projects, I prefer for my POCO objects to be records instead of classes so that they can take advantage of the build in immutablity and value equals.  Please fix.

**Comment by nkosi23** [comment]
Support for record types will be available in Orleans version 3 5.0 according to this recently merged pull request:

https://github.com/dotnet/orleans/pull/7119

However I do not know when 3.5.0 will be released.

**Comment by ReubenBond** [comment]
We are preparing the 3.5.0 release. It's almost ready to be published. We expect to publish it today.
